store/src/java/com/zimbra/cs/account/callback/EphemeralBackendCheck.java

```diff
@@ -58,11 +62,25 @@ public void preModify(CallbackContext context, String attrName,
                 throw ServiceException.FAILURE(String.format(
                         "unable to modify %s; no ephemeral backend specified", attrName), null);
             }
+            savePreviousUrl(context, entry);
+        }
+    }
+
+    private void savePreviousUrl(CallbackContext context, Entry entry) {
+        String prevUrl = ((Config) entry).getEphemeralBackendURL();
+        if (!Strings.isNullOrEmpty(prevUrl)) {
+            context.setData(DataKey.PREV_EPHEMERAL_BACKEND_URL, prevUrl);
         }
     }
 
     @Override
     public void postModify(CallbackContext context, String attrName, Entry entry) {
+        String prevUrl = context.getData(DataKey.PREV_EPHEMERAL_BACKEND_URL);
+        try {
+            ((Config) entry).setPreviousEphemeralBackendURL(prevUrl);
+        } catch (ServiceException e) {
+            ZimbraLog.ephemeral.error(String.format("unable to set zimbraPreviousEphemeralBackendURL to %s", prevUrl), e);
```

Comment:
You don't actually need the String.format here - that is already done by the native method.